LibVLCSharp/Shared/Core.cs

```diff
@@ -64,75 +69,135 @@ static void InitializeAndroid()
-
-                var librariesFolder = Path.Combine(appExecutionDirectory, Constants.LibrariesRepositoryFolderName, arch);
-
-                _libvlccoreHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.CoreLibraryName}.dll");
-                
-                if(_libvlccoreHandle == IntPtr.Zero)
+                bool loadResult;
+                if(PlatformHelper.IsWindows) // TODO: Add linux support
                 {
-                    throw new VLCException($"Failed to load required native library {Constants.CoreLibraryName}.dll");
+                    var libvlccorePath = LibVLCCorePath(libvlcDirectoryPath);
+                    loadResult = LoadNativeLibrary(libvlccorePath, out _libvlccoreHandle);
+                    if(!loadResult)
+                    {
+                        Log($"Failed to load required native libraries at {libvlccorePath}");
+                        return;
+                    }
                 }
 
-                _libvlcHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.LibraryName}.dll");
-                
-                if(_libvlcHandle == IntPtr.Zero)
-                {
-                    throw new VLCException($"Failed to load required native library {Constants.LibraryName}.dll");
-                }
+                var libvlcPath = LibVLCPath(libvlcDirectoryPath);
+                loadResult = LoadNativeLibrary(libvlcPath, out _libvlcHandle);
+                if(!loadResult)
+                    Log($"Failed to load required native libraries at {libvlcPath}");
+                return;
             }
-            else if (PlatformHelper.IsMac)
+
+            if (PlatformHelper.IsLinux) return;
+#if !NETSTANDARD1_1
+            var paths = ComputeLibVLCSearchPaths();
+
+            foreach(var path in paths)
             {
-                _libvlcHandle = PreloadNativeLibrary(appExecutionDirectory, $"{Constants.LibraryName}.dylib");
-                if (_libvlcHandle == IntPtr.Zero)
+                if (PlatformHelper.IsWindows)
                 {
-                    throw new VLCException($"Failed to load required native library {Constants.LibraryName}.dylib");
+                    LoadNativeLibrary(path.libvlccore, out _libvlccoreHandle);
                 }
+                var loadResult = LoadNativeLibrary(path.libvlc, out _libvlcHandle);
+                if (loadResult) break;
             }
-            else if (PlatformHelper.IsLinux)
+
+            if (!Loaded)
             {
-                // Initializes X threads before calling VLC. This is required for vlc plugins like the VDPAU hardware acceleration plugin.
-                if (Native.XInitThreads() == 0)
-                {
-#if !NETSTANDARD1_1
-                    Trace.WriteLine("XInitThreads failed");
+                throw new VLCException($"Failed to load required native libraries. Search paths include {paths.ToString()}");
```

Comment:
`paths.ToString()` won't return what you want here. It will return the type name of `paths`.
You have to do something like that `string.Join(",", paths.Select(p => $"{p.libvlc},{p.libvlccore}"));`

Reply:
Yes you're right. I suspected it, but wanted to wait until I test to get the formatting right. Thanks for the heads up though.